src/dataServices/dynamoDbParamBuilder.ts

```diff
@@ -102,12 +114,25 @@ export default class DynamoDbParamBuilder {
         };
     }
 
-    static buildPutAvailableItemParam(item: any, id: string, vid: number) {
+    /**
+     * Build DDB PUT param to insert a new resource
+     * @param item
+     * @param id
+     * @param vid
+     * @param allowOverwrite - Allow overwriting a resource that already exists
+     * @return DDB params for PUT operation
+     */
+    static buildPutAvailableItemParam(item: any, id: string, vid: number, allowOverwrite: boolean = false) {
         const newItem = DynamoDbUtil.prepItemForDdbInsert(item, id, vid, DOCUMENT_STATUS.AVAILABLE);
-        return {
+        const param: any = {
             TableName: RESOURCE_TABLE,
             Item: DynamoDBConverter.marshall(newItem),
         };
+
+        if (!allowOverwrite) {
+            param.ConditionExpression = 'attribute_not_exists(id)';
+        }

```

Comment:
Is this method used exclusively for CREATEs? if not all new PUTs will fail since puts create a new row with a new vid and the SAME id

Reply:
Yep, it's only being used for CREATEs.

It's only being used here
https://github.com/awslabs/fhir-works-on-aws-persistence-ddb/blob/9ab088450754c74bd50f9768d728cd692cb64000/src/dataServices/dynamoDbDataService.ts#L106